server/src/main/java/org/apache/iotdb/db/query/aggregation/impl/ExtremeAggrResult.java

```diff
@@ -0,0 +1,210 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iotdb.db.query.aggregation.impl;
+
+import org.apache.iotdb.db.query.aggregation.AggregateResult;
+import org.apache.iotdb.db.query.aggregation.AggregationType;
+import org.apache.iotdb.db.query.reader.series.IReaderByTimestamp;
+import org.apache.iotdb.tsfile.exception.write.UnSupportedDataTypeException;
+import org.apache.iotdb.tsfile.file.metadata.enums.TSDataType;
+import org.apache.iotdb.tsfile.file.metadata.statistics.Statistics;
+import org.apache.iotdb.tsfile.read.common.BatchData;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.ByteBuffer;
+
+public class ExtremeAggrResult extends AggregateResult {
+
+  public ExtremeAggrResult(TSDataType dataType) {
+    super(dataType, AggregationType.EXTREME);
+    reset();
+  }
+
+  public Object getAbsValue(Object v) {
+    double doubleValue;
+    float floatValue;
+    int intValue;
+    long longValue;
+
+    switch (resultDataType) {
+      case DOUBLE:
+        doubleValue = (Double) v;
+        return Math.abs(doubleValue);
+      case FLOAT:
+        floatValue = (Float) v;
+        return Math.abs(floatValue);
+      case INT32:
+        intValue = (Integer) v;
+        return Math.abs(intValue);
+      case INT64:
+        longValue = (Long) v;
+        return Math.abs(longValue);
+      default:
+        throw new UnSupportedDataTypeException(String.valueOf(resultDataType));
+    }
+  }
+
+  public Comparable<Object> compareAbsVal(Comparable<Object> minVal, Comparable<Object> maxVal) {
+    Comparable<Object> extVal;
+
+    Comparable<Object> absMaxVal = (Comparable<Object>) getAbsValue(maxVal);
+    Comparable<Object> absMinVal = (Comparable<Object>) getAbsValue(minVal);
+
+    if (absMaxVal.compareTo(absMinVal) >= 0) {
```

Comment:
If the maxVal is 2, minVal is -2, here you choose the maxVal 2 as your defined extremed value which is inconsistent with the following.

Reply:
sorry, I wrote wrong. It isfixed

Reply:
I added IT.I will add docs later